src/main/python/systemml/defmatrix.py

```diff
@@ -504,23 +569,23 @@ def _visit(self, execute=True):
         """
         if self.visited:
             return self
-        self._markAsVisited()
-        if self.data is not None:
-            self._registerAsInput(execute)
+        self._mark_as_visited()
+        if self.eval_data is not None:
+            self._register_as_input(execute)
         elif self.op is not None:
             # Traverse the AST
             for m in self.op.inputs:
                 m._visit(execute=execute)
             self.op._visit(execute=execute)
         else:
             raise Exception('Expected either op or data to be set')
-        if self.data is None and self.output:
-            self._registerAsOutput(execute)
+        if self.eval_data is None and self.output:
+            self._register_as_output(execute)
         return self
 
-    def printAST(self, numSpaces = 0):
+    def _print_ast(self, numSpaces = 0):
```

Comment:
Yeah let's make this function named `print_ast()` since it is intended to be called by the user, and the other variant that is not supposed to be called by the user can still be named `_print_ast(numSpaces)`.